### PR TITLE
Added `scl zoo install` command to install agents at specified paths.

### PR DESCRIPTION
**Pull request by RutvikGupta**
Closes Issue #603.

### CODE REVIEW

File: cli/zoo.py [hunk 1]
```diff
@@ -76,5 +76,45 @@ def manager(port):
...
+        # nargs=-1 in combination with a default value is not supported
+        # if agent_paths is not given, set the known two zoo agent paths as default
+        agent_paths = ["zoo/policies/open-agent", "zoo/policies/rl-agent"]
+
+    for agent_path in agent_paths:
+        pip_install_cmd = [
+            "pip",
+            "install",
+            ".",
+        ]
```
Comment (sah-huawei) [inline cli/zoo.py:101]: This should be outside the loop.

File: cli/zoo.py [hunk 2]
```diff
@@ -76,5 +76,45 @@ def manager(port):
...
+        proc = subprocess.Popen(
+            pip_install_cmd,
+            stderr=subprocess.PIPE,
+            cwd=goal_dir,
+        )
+        proc.wait()
+        proc.communicate()
+
+        if proc.returncode != 0:
+            click.echo(f"{agent_path} already installed")
```
Comment (sah-huawei) [inline cli/zoo.py:113]: Is "already installed" the only reason that the return code might be non-zero?   (For example, what if they don't have internet at the time they try this?)

File: cli/zoo.py (hunk 1 above)
Comment (RutvikGupta) [reply cli/zoo.py:101]: Yeah make sense.

**Review by sah-huawei — APPROVED** [review]

File: cli/zoo.py [hunk 3]
```diff
@@ -76,5 +76,55 @@ def manager(port):
...
+    for agent_path in agent_paths:
+        goal_dir = os.path.join(os.getcwd(), agent_path)
+        clean(goal_dir)
+        proc = subprocess.Popen(
+            pip_install_cmd,
+            stderr=subprocess.PIPE,
+            cwd=goal_dir,
+        )
+        proc.wait()
+        stdout, std_err = proc.communicate()
```
Comment (Adaickalavan) [inline cli/zoo.py:117]: Is `stdout` used?

File: cli/zoo.py [hunk 4]
```diff
@@ -76,5 +76,55 @@ def manager(port):
...
+)
+@click.argument(
+    "agent_paths",
+    type=click.Path(exists=True),
+    metavar="<script>",
+    nargs=-1,
+    required=True,
+)
+def install_agents(agent_paths):
+    if not agent_paths:
```
Comment (Adaickalavan) [inline cli/zoo.py:91]: Why do `if not agent_paths:` when `agent_paths` has `required=True`?

**Review by Adaickalavan — APPROVED** [review]

File: cli/zoo.py [hunk 5]
```diff
@@ -76,5 +76,55 @@ def manager(port):
...
+    ]
+
+    def clean(exec_dir):
+        subprocess.check_call(
+            [sys.executable, "setup.py", "clean", "--all"],
+            cwd=exec_dir,
+        )
+
+    for agent_path in agent_paths:
+        goal_dir = os.path.join(os.getcwd(), agent_path)
```
[the remark this answers is not in the document]
Comment (RutvikGupta) [reply cli/zoo.py]: ok, sure!

File: cli/zoo.py (hunk 3 above)
Comment (RutvikGupta) [reply cli/zoo.py:117]: No, but you need a variable there since `proc.communicate()` returns a tuple.

File: cli/zoo.py (hunk 4 above)
Comment (RutvikGupta) [reply cli/zoo.py:91]: Apparently even is we set `required=True`, the command runs without it so I just added a safety net to be safe.

File: cli/zoo.py (hunk 3 above)
Comment (Adaickalavan) [reply cli/zoo.py:117]: For unused variables, the following will also work: 
```
import subprocess

proc = subprocess.Popen(['echo', 'Hello'], stdout=subprocess.PIPE)
stdout = proc.communicate()[0]
print(stdout)

proc = subprocess.Popen(['echo', 'Hello'], stdout=subprocess.PIPE)
stdout, _ = proc.communicate()
print(stdout)
```

**Pull request merged**